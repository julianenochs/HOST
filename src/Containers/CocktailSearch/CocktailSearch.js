import React, { Component } from 'react';
import { searchCocktailsByName } from '../../apiCalls';
import { cocktailsByName } from '../../Actions/index';
import { connect } from 'react-redux';

export class CocktailSearch extends Component {
  constructor() {
    super();
    this.state = {
      selectedCocktail: ''
    }
  }
  
  handleChange = (e) => {
    this.setState({ selectedCocktail: [e.target.value] })
  }
  
  handleSearch = () => {
    let cocktailQuery = this.state.selectedCocktail
    searchCocktailsByName(cocktailQuery)
      .then(data => {
        this.props.cocktailsByName(data)
      })
  }

  render(){
    return(
      <section>
        <input
          type='search'
          name='search'
          placeholder='Search For Cocktails'
          value={this.state.selectedCocktail}
          onChange={this.handleChange}
          />
          <button onClick={this.handleSearch}> Search </button>
        </section>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  cocktailsByName: (cocktailNames) => dispatch(cocktailsByName(cocktailNames))
});

const mapStateToProps = state => ({
  cocktailsByName: state.cocktailsByName
});

export default connect(mapStateToProps, mapDispatchToProps)(CocktailSearch);