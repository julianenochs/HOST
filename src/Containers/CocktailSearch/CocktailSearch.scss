.cocktail__search {
  width: 40%;
}

.search__input {
  height: 30px;
  font-size: 1.3em;
  width: 75%;
}

.search-cocktail__button {
  background-color: white;
  border: none;
  font-family: 'Federo', sans-serif;
  font-size: 1.5em;
  height: 30px;
  width: 15%;
  &:hover {
      text-shadow: 1px 1px 1px;
  }
}