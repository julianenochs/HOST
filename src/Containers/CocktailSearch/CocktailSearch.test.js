import CocktailSearch from './CocktailSearch';
import React from 'react';
import { shallow } from 'enzyme';

describe('Cocktail Search', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CocktailSearch />)
  });

  it.skip('should match the snapshot', ()=> {
    expect(wrapper).toMatchSnapshot();
  });
});