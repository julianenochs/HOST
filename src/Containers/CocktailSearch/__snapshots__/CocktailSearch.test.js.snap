// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail Search should match the snapshot 1`] = `undefined`;
